pkg/workspace/plugins.go

```diff
@@ -140,18 +142,24 @@ func (info PluginInfo) Install(tarball io.ReadCloser) error {
 
 	// We unpack the tarball into a temporary directory in the plugin cache and after that has completed rename it
 	// to prevent posioning the cache if the install is CTRL-C'd,
-	tempDir := finalDir + ".tmp"
-
-	// If the temporary folder already exists, try to remove it. Note that os.RemoveAll returns an nil error
-	// when the directory does not already exist.
-	if err := os.RemoveAll(tempDir); err != nil {
-		return errors.Wrapf(err, "deleting temporary directory %s", tempDir)
+	//
+	// Generate a random suffix so that two concurrent installs of the same plugin don't stomp over each other.
```

Comment:
Any reason not to use https://golang.org/pkg/io/ioutil/#TempDir?

Reply:
I wanted to control the directory the temporary folder was created in, since I don't want it in `/tmp` so the rename later happens atomically.  But I see you can control the location with TempDir, so I will use this.  Thanks!